.github/workflows/rust.yml

```diff
@@ -11,12 +11,12 @@ env:
 
 jobs:
   check:
-    name: Rust Clippy
+    name: Rust Format, Clippy, and Test
     runs-on: ubuntu-latest
     steps:
-    - uses: actions/checkout@v1
-    - run: rustup component add clippy
-    - uses: actions-rs/clippy-check@v1
-      with:
-        token: ${{ secrets.GITHUB_TOKEN }}
-        args: --all-features
+    - uses: actions/checkout@v4
+    - uses: dtolnay/rust-toolchain@stable
```

Comment:
why though, rust is already installed

Reply:
It won't necessarily be the current version, just whenever GitHub updates their images.